java/analyze-hotspots/src/main/java/com/esri/arcgisruntime/sample/analyzehotspots/MainActivity.java

```diff
@@ -16,316 +16,41 @@
-    // if button is clicked, close the custom dialog
-    analyzeButton.setOnClickListener(new View.OnClickListener() {
-      @Override
-      public void onClick(View v) {
-        analyzeHotspots(fromDateText.getText().toString(), toDateText.getText().toString());
-        dialog.dismiss();
-      }
-    });
-
-    dialog.show();
-  }
-
-  /**
-   * Shows a date picker dialog and writes the date chosen to the correct editable text.
-   *
-   * @param inputCalendar enum which specifies which editable text the chosen date should be written to
-   */
-  private void showCalendar(final InputCalendar inputCalendar) {
-    // create a date set listener
-    DatePickerDialog.OnDateSetListener onDateSetListener = new DatePickerDialog.OnDateSetListener() {
-      @Override public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
-        // build the correct date format for the query
-        StringBuilder date = new StringBuilder()
-            .append(year)
-            .append("-")
-            .append(month + 1)
-            .append("-")
-            .append(dayOfMonth);
-        // set the date to correct text view
-        if (inputCalendar == InputCalendar.From) {
-          fromDateText.setText(date);
-          try {
-            // limit the min date to after from date
-            mMinDate = mSimpleDateFormatter.parse(date.toString());
-          } catch (ParseException e) {
-            e.printStackTrace();
-          }
-        } else if (inputCalendar == InputCalendar.To) {
-          toDateText.setText(date);
-          try {
-            // limit the maximum date to before the to date
-            mMaxDate = mSimpleDateFormatter.parse(date.toString());
-          } catch (ParseException e) {
-            e.printStackTrace();
-          }
-        }
-      }
-    };
-
-    // define the date picker dialog
-    Calendar calendar = Calendar.getInstance();
-    DatePickerDialog datePickerDialog = new DatePickerDialog(MainActivity.this, onDateSetListener,
-        calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH));
-    datePickerDialog.getDatePicker().setMinDate(mMinDate.getTime());
-    datePickerDialog.getDatePicker().setMaxDate(mMaxDate.getTime());
-    if (inputCalendar == InputCalendar.From) {
-      // start from calendar from min date
-      datePickerDialog.updateDate(1998, 0, 1);
-    }
-    datePickerDialog.show();
-  }
-
-  /**
-   * Runs the geoprocessing job, updating progress while loading. On job done, loads the resulting
-   * ArcGISMapImageLayer to the map and resets the Viewpoint of the MapView.
-   *
-   * @param from string which holds a date
-   * @param to   string which holds a date
-   */
-  private void analyzeHotspots(final String from, final String to) {
-    // cancel previous job request
-    if (mGeoprocessingJob != null) {
-      mGeoprocessingJob.cancel();
+    if (savedInstanceState == null) {
```

Comment:
Could there be a case where `savedInstanceState` would not be null? If yes, how would we handle that case?

Reply:
We don't want to re-add the `Fragment` as it's already been retained and restored by the OS. If the `savedInstanceState` isn't null, we're already showing the `Fragment` instance.

Reply:
ok